File: androidunsplash/src/main/java/com/kc/unsplash/api/endpoints/CollectionsEndpointInterface.java
package com.kc.unsplash.api.endpoints;

import com.kc.unsplash.models.Collection;
import com.kc.unsplash.models.Photo;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface CollectionsEndpointInterface {

    @GET("collections")
    Call<List<Collection>> getCollections(@Query("page") Integer page, @Query("per_page") Integer perPage);

    @GET("collections/features")
    Call<List<Collection>> getFeaturedCollections(@Query("page") Integer page, @Query("per_page") Integer perPage);

    @GET("collections/curated")
    Call<List<Collection>> getCuratedCollections(@Query("page") Integer page, @Query("per_page") Integer perPage);

    @GET("collections/{id}")
    Call<Collection> getCollection(@Path("id") String id);

    @GET("collections/curated/{id}")
    Call<Collection> getCuratedCollection(@Path("id") String id);

    @GET("collections/{id}/photos")
    Call<List<Photo>> getCollectionPhotos(@Path("id") String id, @Query("page") Integer page, @Query("per_page") Integer perPage);

    @GET("collections/curated/{id}/photos")
    Call<List<Photo>> getCuratedCollectionPhotos(@Path("id") String id, @Query("page") Integer page, @Query("per_page") Integer perPage);

    @GET("collections/{id}/related")
    Call<List<Collection>> getRelatedCollections(@Path("id") String id);

}

File: androidunsplash/src/main/java/com/kc/unsplash/api/endpoints/CollectionsEndpointInterface.kt
package com.kc.unsplash.api.endpoints

import com.kc.unsplash.models.Photo
import com.kc.unsplash.models.Collection
import com.kc.unsplash.models.SearchResults
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface CollectionsEndpointInterface {

    @GET("collections")
    fun getCollections(@Query("page") page: Int?,
                       @Query("per_page") perPage: Int?): Call<List<Collection>>

    @GET("collections/featured")
    fun getFeaturedCollections(@Query("page") page: Int?,
                               @Query("per_page") perPage: Int?): Call<List<Collection>>

    @GET("collections/curated")
    fun getCuratedCollections(@Query("page") page: Int?,
                              @Query("per_page") perPage: Int?): Call<List<Collection>>

    @GET("collections/{id}")
    fun getCollection(@Path("id") id: String): Call<Collection>

    @GET("collections/curated/{id}")
    fun getCuratedCollection(@Path("id") id: String): Call<Collection>

    @GET("collections/{id}/photos")
    fun getCollectionPhotos(@Path("id") id: String,
                            @Query("page") page: Int?,
                            @Query("per_page") perPage: Int?): Call<List<Photo>>

    @GET("collections/curated/{id}/photos")
    fun getCuratedCollectionPhotos(@Path("id") id: String,
                                   @Query("page") page: Int?,
                                   @Query("per_page") perPage: Int?): Call<List<Photo>>

    @GET("collections/{id}/related")
    fun getRelatedCollections(@Path("id") id: String): Call<List<Collection>>

    @GET("search/collections")
    fun searchCollections(@Query("query") query: String,
                     @Query("page") page: Int,
                     @Query("per_page") perPage: Int): Call<SearchResults<Collection>>
}
